consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/blockcreation/BftMiningCoordinator.java

```diff
@@ -122,7 +124,13 @@ public void awaitStop() throws InterruptedException {
 
   @Override
   public boolean enable() {
-    return true;
+    // Return true if we're already running or idle, or successfully switch to idle
+    if (state.get() == State.RUNNING
+        || state.get() == State.IDLE
+        || state.compareAndSet(State.PAUSED, State.IDLE)) {
```

Comment:
what's the difference between paused and idle?

Reply:
The existing logic had `IDLE`, `RUNNING`, and `STOPPED`, where `IDLE` meant "not started yet". But I wanted to add a "not enabled yet" state where enabling has to happen before starting. So `PAUSED` is intended to mean "Waiting to be enabled, at which point we are IDLE until started".

Reply:
There's an argument that the function calls (`start()`, `stop()`, `enable()`) should move the coordinator into the same state - STARTED, STOPPED, ENABLED etc. but I didn't want to refactor that too much